packages/form-js-viewer/src/core/Validator.js

```diff
@@ -8,10 +10,55 @@ export default class Validator {
 
   validateField(field, value) {
 
-    const { validate } = field;
+    const { type, validate } = field;
 
     let errors = [];
 
+    if (type === 'number') {
+
+      const { decimalDigits, step } = field;
+
+      if (value === 'NaN') {
+
+        errors = [
+          ...errors,
+          'Value is not a number.'
+        ];
+
+      }
+      else if (value) {
+
+        if (decimalDigits !== undefined && decimalDigits >= 0 && countDecimals(value) > decimalDigits) {
```

Comment:
Maybe this is sufficient?

```suggestion
        if (decimalDigits >= 0 && countDecimals(value) > decimalDigits) {
```

Reply:
👍